doc/over_react_redux_documentation.md

```diff
@@ -275,6 +279,221 @@ UiFactory<CounterProps> Counter = connect<CounterState, CounterProps>(
+
+#### Example
+
+> This example assumes that your `Counter` component is rendered as the descendant of a [`ReduxProvider`](#reduxprovider) component
+  that is wired up to a Redux `Store` instance with a `CounterState` instance containing a field called `count`.
+>
+> It also assumes that you have two actions wired up to your reducer - `IncrementAction` and `DecrementAction`.
+
+```dart
+import 'package:over_react/over_react.dart';
+import 'package:over_react/over_react_redux.dart';
+import 'counter_state.dart';
+
+mixin CounterProps on UiProps {}
+
+UiFactory<CounterProps> Counter = uiFunction(
+  (props) {
+    final count = useSelector<int, CounterState>((state) => state.count);
+    final dispatch = useDispatch();
+
+    return Dom.div()(
+      Dom.div()('The current count is $count'),
+      (Dom.button()
+        ..onClick = (_) {
+          dispatch(IncrementAction());
+        }
+      )('+'),
+      (Dom.button()
+        ..onClick = (_) {
+          dispatch(DecrementAction());
+        }
+      )('-'),
+    );
+  },
+  $CounterConfig, // ignore: undefined_identifier
+);
+```
+
+### useStore()
+
+```dart
+Store<TReduxState> useStore<TReduxState>();
+```
+
+Returns a reference to the same Redux `Store` that was passed in to the [`ReduxProvider`](#reduxprovider)
+component that the function component using the hook is a descendant of.
+
+**This hook should probably not be used frequently.** Prefer [`useSelector()`](#useselector) as your primary choice.
+However, this may be useful for less common scenarios that do require access to the `Store`,
+such as replacing reducers.
+
+If you need access to a specific store from a nested [`ReduxProvider`](#reduxprovider) with a custom `Context`,
+> use [`createStoreHook`](#custom-context-for-hooks) instead.
+
+See the [react-redux JS documentation](https://react-redux.js.org/api/hooks#usestore) for more details.
+
+### Custom Context For Hooks
+The [`ReduxProvider`](#reduxprovider) component allows you to specify an alternate `Context` via `props.context`. 
+This is useful if you're building a complex reusable component, and you don't want your store to collide with any 
+Redux store your consumers' applications might use.
+
+To access an alternate context via the hooks API, use the hook creator functions:
+
+```dart
+import 'package:over_react/over_react.dart';
+import 'package:over_react/over_react_redux.dart';
+import 'package:redux/redux.dart';
+
+// ------------------------------------
+//  1. Declare the custom context
+// ------------------------------------
+final MyContext = createContext<MyState>();
+final useMySelector = createSelectorHook(MyContext);
+final useMyDispatch = createDispatchHook(MyContext);
+final useMyStore = createStoreHook(MyContext);
+
+final myStore = Store(myReducer);
+
+// ------------------------------------
+//  2. (Optional) Create a custom
+//  provider component that uses the
+//  context defined above.
```

Comment:
#nit this seems like it provides limited value and adds a bit of boilerplate, and may not be worth the cognitive overhead it adds for the reader (same goes for other examples)

Reply:
Yeah, I was just trying to make our example as close to the react-redux docs as possible.

Reply:
Oh, okay, well if they include it, I suppose that's fine (though I don't agree with them 🙂 )